builtin/logical/database/dbplugin/databasemiddleware.go

```diff
@@ -162,3 +203,76 @@ func (mw *databaseMetricsMiddleware) Close() (err error) {
 	metrics.IncrCounter([]string{"database", mw.typeStr, "Close"}, 1)
 	return mw.next.Close()
 }
+
+// ---- Error Sanitizer Middleware Domain ----
+
+// DatabaseErrorSanitizerMiddleware wraps an implementation of Databases and
+// sanitizes returned error messages
+type DatabaseErrorSanitizerMiddleware struct {
+	l         sync.RWMutex
+	next      Database
+	secretsFn func() map[string]string
```

Comment:
As a future compatibility thing, would it make sense to make this a map[string]interface{} but where all current usages of it are known to be strings? That way if we want to key more complex logic into the sanitizer we'll be able to get more rich data types.